app/src/test/java/com/android/sample/model/network/UserDaoTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.android.sample.model.network
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.room.Room
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.android.sample.model.activity.database.AppDatabase
+import com.android.sample.model.profile.database.UserDao
+import com.android.sample.resources.dummydata.testUser
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
```

Comment:
added tests for the UserDao, including methods to insert and retrieve a user, and to clear users, ensuring the DAO's operations with the local database function correctly.